optimum/neuron/models/training/transformations_utils.py

```diff
@@ -287,7 +443,81 @@ def adapt_state_dict(
                 )
         return state_dict
 
-    def to_original_weights(
+    def _lora_adapt_state_dict(
+        self,
+        module_fully_qualified_name: str,
+        named_parameters: Dict[str, torch.nn.Parameter],
+        orig_state_dict: Dict[str, torch.Tensor],
+        upstanding_sharded_params: Dict[str, torch.Tensor],
+        inplace: bool = False,
+    ) -> Dict[str, torch.Tensor]:
+        tp_size = get_tensor_model_parallel_size()
+        tp_rank = get_tensor_model_parallel_rank()
+
+        if inplace:
+            state_dict = orig_state_dict
+        else:
+            state_dict = dict(orig_state_dict)
+
+        # There are two cases for LoRA:
+        # Case 1: the base layer is a ColumnParallelLinear
+        #   1. "Mix" the LoRA A weights of the several linear layers into a single weight, maybe by doing the mean.
+        #   2. Gather all the LoRA B weights of the linear layers
+        #   3. Shard them across the tensor model parallel size if TP is enabled
+        #   4. Fuse them along the fuse axis
+        # Case 2: the base layer is a RowParallelLinear
+        #   1. "Mix" the LoRA A weights of the several linear layers into a single weight, maybe by doing the mean.
+        #   2. Shard them across the tensor model parallel size if TP is enabled
+        #   3. Gather all the LoRA B weights of the linear layers
+        #   4. Fuse them along the fuse axis
+        fused_linear_fully_qualified_name = f"{module_fully_qualified_name}.{self.fused_linear_name}"
+        param_names = ["weight", "bias"] if self.bias else ["weight"]
+        for param_name in param_names:
+            lora_A_weight_names = [
+                f"{module_fully_qualified_name}.{name}.lora_A.{param_name}" for name in self.linear_names
+            ]
+
+            # Question: how do we "mix" the multiple LoRA A weights?
+            lora_A_weight = torch.mean(
+                torch.stack([state_dict.pop(name) for name in lora_A_weight_names], dim=0),
+                dim=0,
+            )
```

Comment:
In the case where we load lora weights, there is a choice to be made because we will go from having one `lora_A` linear per linear layers, to only one `lora_A` linear for the fused linear layer. So how do we "mix" these `lora_A` weights for the fused layer? Here I decide to do the mean.